src/Context/SoapContextReader.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * @author Sergii Bondarenko, <sb@firstvector.org>
+ */
+namespace Behat\SoapExtension\Context;
+
+use Behat\Testwork\Call\Callee;
+use Behat\Testwork\Environment\Environment;
+use Behat\Testwork\Environment\Reader\EnvironmentReader;
+use Behat\Behat\Context\Reader\ContextReader;
+use Behat\Behat\Context\Environment\ContextEnvironment;
+use Behat\Behat\Context\Environment\UninitializedContextEnvironment;
+
+/**
+ * Class SoapContextReader.
+ *
+ * @package Behat\SoapExtension\Context
+ */
+class SoapContextReader implements EnvironmentReader
+{
+    /**
+     * @var ContextReader[]
+     */
+    private $contextReaders = [];
+    /**
+     * @var Callee[]
+     */
+    private $callees = [];
+
+    /**
+     * Register context loaders.
+     *
+     * @see SoapExtension::process()
+     *
+     * @param ContextReader[] $contextReaders
+     */
+    public function __construct(array $contextReaders)
+    {
+        $this->contextReaders = $contextReaders;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsEnvironment(Environment $environment)
+    {
+        return $environment instanceof ContextEnvironment;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function readEnvironmentCallees(Environment $environment)
+    {
+        if ($environment instanceof UninitializedContextEnvironment) {
```

Comment:
Ok, agreed for now, we'll see how it goes, we only have 1 context now, but having it automatically included is a good feature I think